### PR TITLE
Regenerate the FFI test expectations

### PR DESCRIPTION
**Pull request by kmillikin**
We've started passing on of the failing tests, and we fail some of the others in slightly different ways.

### CODE REVIEW

File: ffi_tests_output.txt [hunk 1]
```diff
@@ -311,29 +301,14 @@ Running tests/ffi/regress_jump_to_frame_test.dart
 
 
 ===== CRASH =====
-si_signo=Bus error: 10(10), si_code=1, si_addr=0x600002058e70
-version=3.0.6 (stable) (Fri Jul 21 14:08:38 2023 -0700) on "macos_simarm64"
-pid=51357, thread=11267, isolate_group=main(0x13d01aa00), isolate=main(0x13d02a800)
+si_signo=Segmentation fault: 11(11), si_code=2, si_addr=0x10
```
Comment (eseidel) [inline ffi_tests_output.txt:304]: Interesting that we're now dereffing null (well offset 0x10 from zero).

File: ffi_tests_output.txt [hunk 2]
```diff
@@ -406,31 +381,29 @@ Passed: tests/ffi/vmspecific_function_callbacks_test.dart
 Compiling tests/ffi/vmspecific_function_gc_test.dart
 Running tests/ffi/vmspecific_function_gc_test.dart
 
-../../runtime/vm/simulator_arm64.cc: 955: error: expected: instr == NULL || reg != R18
-version=3.0.6 (stable) (Fri Jul 21 14:08:38 2023 -0700) on "macos_simarm64"
-pid=51474, thread=15875, isolate_group=main(0x13d812200), isolate=main(0x13d81f400)
+Simulator hit Unimplemented instruction: at 0x102c25b7c, last_pc=0x102dcfb9c
```
Comment (eseidel) [inline ffi_tests_output.txt:384]: Ended up in clang code?

File: ffi_tests_output.txt [hunk 3]
```diff
@@ -449,55 +422,33 @@ testHandleWithInteger
...
 testReturnHandleInCallback
-ReturnHandleInCallback 0x1042324e8
+ReturnHandleInCallback 0x102422af8
 returnHandleCallback returning Instance of 'SomeClass'
 HandleReadFieldValue
 Dart_PropagateError(field_value)
 
 
 ===== CRASH =====
 si_signo=Segmentation fault: 11(11), si_code=2, si_addr=0x10
```
Comment (eseidel) [inline ffi_tests_output.txt:444]: Interesting that this old crash matches the new crash in signature.

**Review by eseidel — APPROVED** [review]

File: ffi_tests_output.txt (hunk 1 above)
Comment (kmillikin) [reply ffi_tests_output.txt:304]: This is another corner case: there is no transition `next` above the handler frame so we can't unwind down to `next->previous()` and instead will have to effectively unwind down to the top transition.

File: ffi_tests_output.txt (hunk 2 above)
Comment (kmillikin) [reply ffi_tests_output.txt:384]: Yes, it looks like it.  We are calling native code through the ExitSafepoint stub.  We either need to ensure there's a "to CPU" transition in that stub, or that the stub itself is run on the CPU.

**Pull request merged**